tide/master/StateSerializationHelper.cpp

```diff
@@ -66,6 +66,20 @@ bool _canBeRestored( const CONTENT_TYPE type )
     return true;
 }
 
+void _relocateTempContent( DisplayGroup& group, const QString& uploadDir )
+{
+    for ( const auto& window : group.getContentWindows( ))
+    {
+        const auto& uri = window->getContent()->getURI();
+        if( uri.startsWith( QDir::tempPath( )))
+        {
+            const auto newUri = uploadDir + "/" +  QFileInfo( uri ).fileName();
+            QDir().rename( uri, newUri );
```

Comment:
The files should be placed in a subfolder with the session name, otherwise a second file with the same name will overwrite the one from a previous session:
uploads/sessionABC/image.png
uploads/sessionXYZ/image.png
Also, log a message if the rename operation fails for some reason and don't replace the window's content...